grails-app/controllers/org/bbop/apollo/OrganismController.groovy

```diff
@@ -113,6 +151,462 @@ class OrganismController {
+                        if (trackService.findTrackFromArray(tracksArray, trackConfigObject.get("label")) == null) {
+                            // add track config to trackList.json
+                            tracksArray.add(trackConfigObject)
+                            // unpack track data into organism directory
+                            File zipFile = new File(trackDataFile.getOriginalFilename())
+                            trackDataFile.transferTo(zipFile)
+                            try {
+                                // TODO: how to differentiate between which track data goes to tracks folder and which stays at directory root
+                                organismService.unzip(zipFile, newDirectoryName + File.separator + "tracks", trackConfigObject.get("label"), true)
+                                def trackListJsonWriter = trackListJsonFile.newWriter()
+                                trackListJsonWriter << trackListObject.toString(4)
+                                trackListJsonWriter.close()
+                            }
+                            catch (IOException e) {
+                                returnObject.getJSONArray("status").add(e.message)
+                                log.error e.printStackTrace()
+                            }
+                        }
+                        else {
+                            log.error "an entry for track ${trackConfigObject.get("label")} already exists in ${newDirectoryName}${File.separator}trackList.json"
+                            returnObject.getJSONArray("status").add("an entry for track ${trackConfigObject.get("label")} already exists in ${organism.directory}${File.separator}trackList.json")
+                        }
+                    }
+
+                }
+            } else {
+                returnObject.getJSONArray("status").add("Organism not found")
+                log.error "Organism not found"
+            }
+
+        } catch (e) {
+            returnObject.getJSONArray("status").add(e.message)
+            log.error e.message
+        }
+
+        render returnObject as JSON
+    }
+
+    @RestApiMethod(description = "Deletes a track, previously added via web services, from an existing organism", path = "/organism/deleteTrackFromOrganism", verb = RestApiVerb.POST)
+    @RestApiParams(params = [
+            @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "organism", type = "string", paramType = RestApiParamType.QUERY, description = "An id or commonName used to uniquely identify the organism")
+            , @RestApiParam(name = "trackLabel", type = "string", paramType = RestApiParamType.QUERY, description = "Track label corresponding to the track that is to be deleted")
+    ])
+    @Transactional
+    def deleteTrackFromOrganism() {
```

Comment:
I wonder if we should include an "updateTrackForOrganism()"   I'm not sure if that would be exactly common.   If nothing else, not very necessary at this point in time.